# Домашнє завдання до теми "Основи роботи з рядками та математичні операції"

1. Створіть змінні `a`, `b`, `c` та `d`, задайте їм значення відповідно 101, 210, 111, 402, та обчисліть з допомогою Python наступний вираз:  $$ \frac{(a+b) * c}{d} $$

**Важливо**: тут ми маємо обчислити значення саме з використанням змінних, а не просто чисел, як би ми робили у калькуряторі. Змінні потрібні аби за бажання ми могли змінити їх значення та не переписувати код обчислення виразу, бо він буде працювати для різних значень змінних.

In [1]:
a = 101
b = 210
c = 111
d = 402
result = ((a + b ) * c) / d
print(result)

85.8731343283582


Очікувана відповідь: 85.8731343283582

2. Якою буде остача від ділення числа 11283 на 89? Обчисліть з допомогою Python.

In [4]:
num1 = 11283
num2 = 89
remainder = num1 % num2 
print( remainder)

69


3. Скільки цілих разів поміститься число 89 в 11283? (Або що те саме: який буде результат цілочисельного ділення 11283 на 89?)

In [5]:
int_num = num1 // num2 
print(int_num)

126


4. Запишіть у змінну `name` ваше імʼя.

In [6]:
name = 'Denys'

5. Запишіть у змінну `surname` ваше прізвище.

In [7]:
surname = "Horkavyi"

6. Сконкатенуйте імʼя та прізвище в одну фразу через пробіл. Запишіть результат у нову змінну `name_surname` та виведіть результат на екран.  

In [10]:
name_surname  =  name + ' ' + surname
name_surname

'Denys Horkavyi'

Очікувана відповідь (тільки з вашими даними): 'Hanna Pylieva'

7. Зробіть усі літери змінної `name_surname` заглавними використовуючи метод рядків в Python:

In [12]:
name_surname.upper()

'DENYS HORKAVYI'

Очікувана відповідь (тільки з вашими даними): 'HANNA PYLIEVA'

8. Виведіть третю літеру вашого `name_surname`. Тут уважно, треба саме третя літера. Згадайте, з якого значення починається індексування значень в Python.

In [13]:
display(name_surname[2])

'n'

Очікувана відповідь (тільки з вашими даними): 'n'

9. Виведіть першу та останню літеру вашого прізвища.

In [14]:
display(surname[0], surname[-1])

'H'

'i'

Очікувана відповідь (тільки з вашими даними): ('P', 'a')

10. Виведіть довжину вашого прізвища.

In [16]:
print(len(surname))

8


Очікувана відповідь (тільки з вашими даними): 7


11. Виведіть кількість літер `a` у столиці Мадагаскару `Antananarivo`. Важливо врахувати всі літери, як великі, так і маленькі.

In [17]:
madagascar_cap = 'Antananarivo'
count = madagascar_cap.lower().count('a')
display(count)

4

Очікувана відповідь: 4

12. Створіть рядок `my_string` зі значенням: "The quick brown fox jumps over the lazy dog!!!".   
Перевірте, чи починається рядок `my_string` з "The" і чи закінчується він "dog!". Виведіть результати на екран.

In [18]:
my_string = "The quick brown fox jump over the lazy dog!!!"
# почааток рядка
if my_string.startswith("The"):
    print("Рядок починається з 'The'")
else:
    print("Рядок НЕ починається з 'The'")
# кинець рядка 
if my_string.endswith("dog!"):
    print("Рядок закінчується на 'dog!'")
else:
    print("Рядок НЕ закінчується на 'dog!'")

Рядок починається з 'The'
Рядок НЕ закінчується на 'dog!'


13. Розділіть рядок `my_string` на список окремих слів, а потім з'єднайте всі слова через нижнє підкреслювання. Виведіть результат на екран.

In [21]:
new_string = '_'.join(my_string.split())
display(new_string)

'The_quick_brown_fox_jump_over_the_lazy_dog!!!'

Очікуваний результат: `The_quick_brown_fox_jumps_over_the_lazy_dog!!!`